query-engine/connectors/sql-query-connector/src/database/operations/coerce.rs

```diff
@@ -69,16 +77,20 @@ fn coerce_json_relation_to_pv(value: serde_json::Value, rs: &RelationSelection)
             let related_model = rs.field.related_model();
 
             for (key, value) in obj {
-                match related_model.fields().all().find(|f| f.db_name() == key).unwrap() {
-                    Field::Scalar(sf) => {
+                match related_model.fields().all().find(|f| f.db_name() == key) {
+                    Some(Field::Scalar(sf)) => {
                         map.push((key, coerce_json_scalar_to_pv(value, &sf)?));
                     }
-                    Field::Relation(rf) => {
+                    Some(Field::Relation(rf)) => {
                         // TODO: optimize this
                         if let Some(nested_selection) = relations.iter().find(|rs| rs.field == rf) {
                             map.push((key, coerce_json_relation_to_pv(value, nested_selection)?));
                         }
                     }
+                    None => {
+                        let coerced_value = coerce_json_virtual_field_to_pv(&key, value)?;
+                        map.push((key, coerced_value));
+                    }
```

Comment:
Can this `None` case actually happen at runtime?

Reply:
Yes, for `_count` 😄

Reply:
`None` here means that the field does not exist on the model, and virtual fields don't

Reply:
This is a sign we're dealing with the wrong abstractions. Works fine atm because we only have virtuals, but if we ever have something else, we'll be screwed. It's ok for now, I'd still very much like to defer all of this in the serializer, where we will have access to the query schema and not just the model.

Reply:
Yep, this whole module kind of feels like a part of serializer that leaked into a different level.